Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs

```diff
@@ -37,18 +37,20 @@ public Session CreateSession(DatabaseContext _db, User user)
         public Session ValidateAndUpdateSession(DatabaseContext _db, string token)
         {
             Session response = _sessionService.GetSession(_db, token);
-
-            if(response != null)
-            {
-                return _sessionService.UpdateSession(_db, response);
-            }
-            else
-            {
-                return null;
-            }
-        }
-
-        public Session DeleteSession(DatabaseContext _db, string token)
+			//For the Compare method, a -1 is returned if the current Session expiration is shorter than the current time
+			//A 0 is returned if they are equal
+			//A 1 is returned if current Session is longer than the current time
+			if (TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == -1 || TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == 0)
```

Comment:
what about the case of someone starting the session at 11:50pm? and the check it done after midnight? just wondering

Reply:
I will look into it. Im pretty sure the TimeSpan.Compare method will take care of it. Will update when I have an answer.